.travis.yml

```diff
@@ -10,32 +11,48 @@ before_install:
   - cat /proc/cpuinfo | grep cores | wc -l
   - free -h
 install:
-  - hibench=$(pwd)
-  - cd /opt/
-  - wget http://d3kbcqa49mib13.cloudfront.net/spark-1.6.0-bin-hadoop2.6.tgz
-  - tar -xzf spark-1.6.0-bin-hadoop2.6.tgz
-  - wget https://archive.apache.org/dist/hadoop/core/hadoop-2.6.5/hadoop-2.6.5.tar.gz
-  - tar -xzf hadoop-2.6.5.tar.gz
-  - cd ${hibench}
-  - cp ./travis/spark-env.sh /opt/spark-1.6.0-bin-hadoop2.6/conf/
-  - cp ./travis/core-site.xml /opt/hadoop-2.6.5/etc/hadoop/
-  - cp ./travis/hdfs-site.xml /opt/hadoop-2.6.5/etc/hadoop/
-  - cp ./travis/mapred-site.xml /opt/hadoop-2.6.5/etc/hadoop/
-  - cp ./travis/yarn-site.xml /opt/hadoop-2.6.5/etc/hadoop/
-  - cp ./travis/hibench.conf ./conf/
-  - cp ./travis/benchmarks.lst ./conf/
+  - |
+    jc_ver="$(javac -version 2>&1| cut -d ' ' -f2)"
+    if [[ "$jc_ver" =~ 11 ]]; then
+        echo "No Hadoop and Spark config implementation for ${jc_ver} yet."
+    elif [[ "$jc_ver" =~ 8 ]]; then
+        hibench=$(pwd)
+        cd /opt/
+        wget http://d3kbcqa49mib13.cloudfront.net/spark-1.6.0-bin-hadoop2.6.tgz
+        tar -xzf spark-1.6.0-bin-hadoop2.6.tgz
+        wget https://archive.apache.org/dist/hadoop/core/hadoop-2.6.5/hadoop-2.6.5.tar.gz
+        tar -xzf hadoop-2.6.5.tar.gz
+        cd ${hibench}
+        cp ./travis/spark-env.sh /opt/spark-1.6.0-bin-hadoop2.6/conf/
+        cp ./travis/core-site.xml /opt/hadoop-2.6.5/etc/hadoop/
+        cp ./travis/hdfs-site.xml /opt/hadoop-2.6.5/etc/hadoop/
+        cp ./travis/mapred-site.xml /opt/hadoop-2.6.5/etc/hadoop/
+        cp ./travis/yarn-site.xml /opt/hadoop-2.6.5/etc/hadoop/
+        cp ./travis/hibench.conf ./conf/
+        cp ./travis/benchmarks.lst ./conf/
+        echo "export JAVA_HOME=${JAVA_HOME}" >> /opt/hadoop-2.6.5/etc/hadoop/hadoop-env.sh
+    fi
 before_script:
   - "export JAVA_OPTS=-Xmx512m"
 cache:
   directories:
   - $HOME/.m2
 script:
-  - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=2.2 -Dscala=2.11
-  - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=2.0 -Dscala=2.11
-  - mvn clean package -q -Dmaven.javadoc.skip=true -Dspark=1.6 -Dscala=2.10
-  - sudo -E ./travis/configssh.sh
-  - sudo -E ./travis/restart_hadoop_spark.sh
-  - cp ./travis/hadoop.conf ./conf/
-  - cp ./travis/spark.conf ./conf/
-  - /opt/hadoop-2.6.5/bin/yarn node -list 2
-  - sudo -E ./bin/run_all.sh
+  - |
+    jc_ver="$(javac -version 2>&1| cut -d ' ' -f2)"
+    if [[ "$jc_ver" =~ 11 ]]; then
+        mvn clean package -q -Psparkbench -Pflinkbench -Phadoopbench -Pstormbench -Dmaven.javadoc.skip=true -Dhadoop=3.2 -Dspark=2.4 -Dscala=2.12
```

Comment:
We also need  `sudo -E ./bin/run_all.sh` here

Reply:
`run_all.sh` executed for all of those JDK.
So it's tested 7, 8 and 11.